packages/itwin/grouping-mapping-widget/src/widget/components/Mappings.tsx

```diff
@@ -0,0 +1,115 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import React, { useCallback, useEffect, useState } from "react";
+import {
+  handleError,
+} from "./utils";
+import { useMappingClient } from "./context/MappingClientContext";
+import type { IMappingsClient, Mapping } from "@itwin/insights-client";
+import type { GetAccessTokenFn, GroupingMappingApiConfig } from "./context/GroupingApiConfigContext";
+import { useGroupingMappingApiConfig } from "./context/GroupingApiConfigContext";
+import type { CreateTypeFromInterface } from "../utils";
+import type { mappingUIDefaultDisplayStrings } from "./MappingsUI";
+import { MappingsUI } from "./MappingsUI";
+
+export type IMappingTyped = CreateTypeFromInterface<Mapping>;
+
+export interface MappingsProps {
+  onClickAddMapping?: () => void;
+  onClickMappingTitle?: (mapping: Mapping) => void;
+  onClickMappingModify?: (mapping: Mapping) => void;
+  displayStrings?: Partial<typeof mappingUIDefaultDisplayStrings>;
+}
+
+const fetchMappings = async (
+  setMappings: React.Dispatch<React.SetStateAction<Mapping[]>>,
+  iModelId: string,
+  setIsLoading: React.Dispatch<React.SetStateAction<boolean>>,
+  getAccessToken: GetAccessTokenFn,
+  mappingsClient: IMappingsClient
+) => {
+  try {
+    setIsLoading(true);
+    const accessToken = await getAccessToken();
+    const mappings = await mappingsClient.getMappings(accessToken, iModelId);
+    setMappings(mappings.sort((a, b) => a.mappingName.localeCompare(b.mappingName)));
+  } catch (error: any) {
+    handleError(error.status);
+  } finally {
+    setIsLoading(false);
+  }
+};
+
+export interface MappingsOperationsProps extends GroupingMappingApiConfig {
+  mappingClient: IMappingsClient;
+}
+
+export const useMappingsOperations = ({ iModelId, getAccessToken, mappingClient }: MappingsOperationsProps) => {
```

Comment:
I would consider putting this hook in its own file, perhaps "src/hooks/useMappingsOperations.ts".

Reply:
Sure, that was the plan in the next PR. There will be some file organization.